app/javascript/components/select/Selector.vue

```diff
@@ -0,0 +1,200 @@
+            option.hasOwnProperty("label") &&
+            option.hasOwnProperty("value")
+          )
+            continue;
+          return false;
+        }
+        return true;
+      }
+    }
+  },
+  created() {
+    document.documentElement.addEventListener("click", this.onDocumentClick);
+    if (typeof this.default !== "undefined" && !this.state) {
+      this.select(this.default);
+    }
+  },
+  beforeDestroy() {
+    document.documentElement.removeEventListener("click", this.onDocumentClick);
+  },
+  data() {
+    return {
+      dropdownEnabled: false,
+      search: undefined,
+      searchClear: undefined
+    };
+  },
+  computed: {
+    computedOptions() {
+      return [
+        {
+          value: undefined,
+          label: ""
+        },
+        ...this.options.map(opt => {
+          if (typeof opt === "string") return { value: opt, label: opt };
+          return {
+            value: opt.value,
+            label: opt.label
+          };
+        })
+      ];
+    },
+    defaultOption() {
+      return (
+        this.computedOptions.filter(
+          option => option.value === this.default
+        )[0] || this.computedOptions[0]
+      );
+    },
+    optionSelected() {
+      try {
+        return (
+          this.computedOptions.filter(opt => opt.value === this.state)[0] ||
+          this.defaultOption
+        );
+      } catch (e) {
+        console.error(e);
+      }
+      return this.defaultOption;
+    }
+  },
+  watch: {
+    dropdownEnabled(dropdownEnabled, previousDropdownEnabled) {
+      if (dropdownEnabled && previousDropdownEnabled === false) {
+        // focus the first item when the dropdown is toggled
+        delay(() => this.$refs.options[0].focus(), 0); // delay fixes issue with focus
+      }
+    }
+  },
+  methods: {
+    onDocumentClick(e) {
+      e.preventDefault();
+      e.stopPropagation();
+      if (this.$el.contains(document.activeElement)) {
+      } else {
+        this.showDropdown(false);
+      }
+    },
+    onKeyup(e) {
+      if (this.searchClear) {
+        // if there's a timeout ID, clear it
+        clearTimeout(this.searchClear);
+      }
+      // treat keyboard input keys as literal keys if their length=1 & ignore everything else
+      if (e.key.length === 1) {
+        this.search = this.search ? this.search + e.key : e.key;
+        const searchResult = this.computedOptions.slice(1).filter(option =>
+          // when the search is length=1, match from the beginning of the labels, else match against whole label
+          new RegExp(
+            (this.search.length === 1 ? "^" : "") + this.search,
+            "i"
+          ).test(option.label)
+        )[0];
+        if (searchResult) {
+          if (this.dropdownEnabled) {
+            // focus if you have the options in front of you to choose from
+            this.$refs.options[
+              this.computedOptions.slice(1).indexOf(searchResult)
+            ].focus();
```

Comment:
How well does this work if the dropdown has a scroll?

Reply:
Works fine as far as I can see, are any particular concerns you were thinking of?

Reply:
The page can sometimes jump around weirdly when focusing an object, rather than just scrolling within the options container.

Reply:
Ah I see, I don't think it matters because it's in an overflow-y scroll, but it is difficult to prevent body scroll if using the arrow keys to tab between the options (I did think about looking into it), but decided against this because of time

Reply:
Shouldn't be a problem here, as it's a short list anyway, but worth thinking about if it gets longer in the future. I think there's something in O+ habitats that scrolls acceptably (if not amazingly!)